app/src/main/java/com/nutomic/syncthingandroid/activities/SettingsActivity.java

```diff
@@ -365,22 +369,26 @@ public boolean onSyncthingPreferenceChange(Preference preference, Object o) {
                         mOptions.urAccepted = ((boolean) o)
                                 ? systemInfo.urVersionMax
                                 : Options.USAGE_REPORTING_DENIED;
-                        mApi.editSettings(mGui, mOptions, getActivity());
                     });
                     break;
                 default: throw new InvalidParameterException();
             }
 
             mApi.editSettings(mGui, mOptions, getActivity());
+            mRequireRestart = true;
```

Comment:
This is a misunderstanding caused by the behaviour of the original restart code. Nutomic (and other contributors) made this orginally that the user changes a setting, is told to restart and if the user does confirm, then(!) the new syncthing config gets posted and syncthing gets restarted after the post request completed. Now, we just queue restart requests until the user "leaves" the settings Ui and then do the same procedure. To sum up: syncthing core will first know that the config should change on the POST request and could then issue a restart request - but - it is already restarted by the existing code behaviour.
I agree we can improve this further in a follow up PR. The suggestion to post the config early to syncthing and check for the web UI "restart box" would also make sense, but not in this PR.

Reply:
Right, so syncthing is restarted unconditionally, even if the config posted does not require a restart, because it's changing something like discovery addresses which we can handle online?

Reply:
Oh, not sure if pointed out correctly what would happen if mRequireRestart = true would be left out on change of a specific option: So, if the line wouldn't be there for example at "maxSendKbps" the config would never get posted to syncthing and on the next change that occurs from the wrapper, e.g. a new folder is created, this change would be picked up. BUT - if after this "is not posted" the user gets into the web UI and changes some setting, the restart behaviour of the event processor would cause a re-read of the new syncthing core config and discard the "in wrapper's memory cache" config holding the maxSendkbps change > the change would go lost.
So, again, I can underline there are quite many problems in the existing implementation throughout the wrapper that need a good cleanup strategy and multiple small PR in order to not to f*** things up. Not intending to offend anyone as I'm sure the original code writers had good reason to do so - if we would start with this app from scratch (only mind play) I would have suggested doing everything at runtime through the REST API and sticking as close as possible to how the javascript of the web UI does it. Currently, syncthing android has many implementations from my point of view, where it does read current config to a cache, modify the cache, write it back using a "fully converted cache"  strategy. In between cache commits from apps UI code and syncthing's REST/Events interface occur and sometimes result in a restart to "synchronize the one cache written by multiple accessors" and sometimes not. It's always a long time of code reading for me to understand what of the strategies was applied in the past to accomplish a config task.
Please go easy on me as I just wanna help. I'm also not responsible for some code parts that need improvement but one day we'll get 'em.

Reply:
Ok, so RequireRestart is more like ConfigChanged